Translate otlp metrics to prom metatada (#5693)
grafana/mimir: CHANGELOG.md

* [ENHANCEMENT] Query-frontend: add `cortex_query_frontend_enqueue_duration_seconds` metric that records the time taken to enqueue or reject a query request when not using the query-scheduler. #5879
* [ENHANCEMENT] Expose `/sync/mutex/wait/total:seconds` Go runtime metric as `go_sync_mutex_wait_total_seconds_total` from all components. #5879
* [ENHANCEMENT] Go: updated to 1.21.1. #5955

> * [ENHANCEMENT] Distributor: store metadata when ingesting metrics via OTLP. This makes metrics description and type available when ingest OTLP protocol. #5693

Reviewer: ```suggestion
* [ENHANCEMENT] Distributor: store metadata when ingesting metrics via OTLP. This makes metrics description and type available when ingesting metrics via OTLP. #5693
```

> * [ENHANCEMENT] Distributor: store metadata when ingesting metrics via OTLP. This makes metrics description and type available when ingest OTLP protocol. #5693

Reviewer: ```suggestion
* [ENHANCEMENT] Distributor: add experimental support for storing metadata when ingesting metrics via OTLP. This makes metrics description and type available when ingest OTLP protocol. Enable with `-distributor.enable-otlp-metadata-storage=true`. #5693
```

> * [ENHANCEMENT] Distributor: store metadata when ingesting metrics via OTLP. This makes metrics description and type available when ingesting metrics via OTLP. #5693

Reviewer: ```suggestion
* [ENHANCEMENT] Distributor: add experimental support for storing metadata when ingesting metrics via OTLP. This makes metrics description and type available when ingesting metrics via OTLP. Enable with `-distributor.enable-otlp-metadata-storage=true`. #5693
```

* [ENHANCEMENT] Ingester: optimize label-values with matchers call when number of matched series is small. #5600
* [ENHANCEMENT] Compactor: Delete bucket-index, markers and debug files if there are no blocks left in the bucket index. This cleanup must be enabled by using `-compactor.no-blocks-file-cleanup-enabled` option. #5648
* [ENHANCEMENT] Ingester: reduce memory usage of active series tracker. #5665

> * [ENHANCEMENT] Distributor: Translate OTLP to Metadata and forward it. This makes metrics description and type available when ingest OTLP protocol. #5693

Reviewer: ```suggestion
* [ENHANCEMENT] Distributor: store metadata when ingesting metrics via OTLP. This makes metrics description and type available when ingest OTLP protocol. #5693
```
